srcts/src/shiny/bind.ts

```diff
@@ -61,31 +61,49 @@ const bindingsRegistery = (() => {
   const bindings: IdToBindingTypes = new Map();
 
   /**
-   * Checks if the bindings registery is valid. Currently this just checks for
+   * Checks if an ID is already registered to an input or output binding
+   * @param id ID of the possibly bound input or output
+   * @param type Type of binding to check for, either "input", "output", or
+   * "all".
+   * @returns true if the ID is already registered to a binding, otherwise false
+   */
+  function isRegistered(
+    id: string,
+    type: "all" | "input" | "output" = "all"
+  ): boolean {
+    const record = bindings.get(id);
+    if (!record) return false;
+    if (type === "all") return true;
+    return record.includes(type);
+  }
+
+  /**
+   * Checks if the bindings registry is valid. Currently this just checks for
    * duplicate IDs but in the future could be expanded to check more conditions
-   * @returns ShinyClientError if current ID bindings are invalid, otherwise null
+   * @returns ShinyClientError if current ID bindings are invalid, otherwise
+   * returns an ok status.
    */
   function checkValidity():
     | { status: "error"; error: ShinyClientError }
     | { status: "ok" } {
-    const duplicateIds: IdToBindingTypes = new Map();
+    const duplicateIds: { [id: string]: { input: number; output: number } } =
+      {};
```

Comment:
Curious as to why we're switching away from `Map` here? Maps tend to be the best choice for collections of unknown sizes.

Reply:
I switched it back to a map in 616bda08bc823b95e37e6afc53545008a34a3ae6